Add support for CockroachDB StateStore

r-c-correa opened the pull request:
## Description
Add Query support for  CockroachDB StateStore

## Issue reference
#2236

msfussell committed:
security concepts update

Signed-off-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

vfarcic committed:
Video (#2198)

Signed-off-by: Viktor Farcic <viktor@farcic.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

pravinpushkar committed:
Updating request payload format for cosmosdbgremlinapi (#2161)

Signed-off-by: Pravin Pushkar <ppushkar@microsoft.com>

Co-authored-by: Mark Fussell <mfussell@microsoft.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

siredmar committed:
Update setup-jetstream.md (#2200)

* Update setup-jetstream.md

Added missing information how to create the JetStream. Creating a JetStream is essential. Otherwise the pub/sub for jetstream won't work properly.

Signed-off-by: Armin Schlegel <armin.schlegel@gmx.de>

* Update setup-jetstream.md

Signed-off-by: Armin Schlegel <armin.schlegel@gmx.de>

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

berndverst committed:
Update mTLS certificate rotation guidance (#2208)

* Update mTLS rotation guidance

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update daprdocs/content/en/operations/security/mtls.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update daprdocs/content/en/operations/security/mtls.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update daprdocs/content/en/operations/security/mtls.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Update daprdocs/content/en/operations/security/mtls.md

Co-authored-by: Yaron Schneider <schneider.yaron@live.com>
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

* Refine downtime warning

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Co-authored-by: Yaron Schneider <schneider.yaron@live.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

berndverst committed:
Remove unnecessary sentry restarts (#2209)

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

Taction committed:
Add doc for pubsub rabbitmq exchage kind (#2207)

* Add doc for pubsub rabbitmq exchage kind

Signed-off-by: zhangchao <zchao9100@gmail.com>

* fix review

Signed-off-by: zhangchao <zchao9100@gmail.com>

* fix review

Signed-off-by: zhangchao <zchao9100@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

dmitsh committed:
fix redirect error (#2218)

Signed-off-by: Dmitry Shmulevich <dmitsh@users.noreply.github.com>

Co-authored-by: Dmitry Shmulevich <dmitsh@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

dmitsh committed:
add docs for querying Redis state store (#2216)

Signed-off-by: Dmitry Shmulevich <dmitsh@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

duglin committed:
Update get-started-component.md (#2223)

Teach them about "dapr stop" while we're at it

Signed-off-by: Doug Davis <dug@microsoft.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

hhunter-ms committed:
--dapr-http-port fix (#2212)

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

Co-authored-by: Mark Fussell <mfussell@microsoft.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
Update to security docs

Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

RichardLaos committed:
Update autoDeleteOnIdleInSec field description (#2211)

Signed-off-by: Richard Laos <richard.laos@quest.com>

Co-authored-by: Mark Fussell <mfussell@microsoft.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

gehhilfe committed:
Jetstream statestore and auth (#2204)

* Add example for nats jetstream with nkey based authentication

Signed-off-by: Tim Burkert <burkert.tim@gmail.com>

* Add links to nats jwt authentication

Signed-off-by: Tim Burkert <burkert.tim@gmail.com>

* Add component reference for jetstream state store

Signed-off-by: Tim Burkert <burkert.tim@gmail.com>

* Update daprdocs/content/en/reference/components-reference/supported-state-stores/setup-jetstream-kv.md

Co-authored-by: Mark Fussell <markfussell@gmail.com>

* Improve jetstream kv state store documentation

Signed-off-by: Tim Burkert <burkert.tim@gmail.com>

* Changing cli to CLI

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
updating overview

Signed-off-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
updates to diagrams

Signed-off-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
Update daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md

Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
Update daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md

Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
Update daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md

Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
Update daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-namespaces.md

Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
Update daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-overview.md

Co-authored-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
Update daprdocs/content/en/operations/components/component-scopes.md

Co-authored-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
Update daprdocs/content/en/operations/components/component-scopes.md

Co-authored-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
Update daprdocs/content/en/operations/components/component-scopes.md

Co-authored-by: Hannah Hunter <94493363+hhunter-ms@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

duglin committed:
Add a missing back-tick (#2227)

Signed-off-by: Doug Davis <dug@microsoft.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

berndverst committed:
Add component reference to AAD authentication

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

berndverst committed:
Also add blobstorage binding AAD support

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

berndverst committed:
no please sir

Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
adding video

Signed-off-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell committed:
fixed link to wrong image

Signed-off-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

r-c-correa committed:
support for CockroachDB StateStore

Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

hhunter-ms committed:
add missing changes (#2239)

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>

hhunter-ms committed:
add missing changes (#2239)

Signed-off-by: Hannah Hunter <hannahhunter@microsoft.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cockroachdb.md:
```diff
@@ -0,0 +1,74 @@
+
+Create a file called `cockroachdb.yaml`, paste the following and replace the `<CONNECTION STRING>` value with your connection string. The connection string for CockroachDB follow the same standard for PostgreSQL connection string. For example, `"host=localhost user=root port=26257 connect_timeout=10 database=dapr_test"`. See the CockroachDB [documentation on database connections](https://www.cock ...
+
+If you want to also configure CockroachDB to store actors, add the `actorStateStore` configuration element shown below.
```
Given that the overview describes that this state store supports actors, you do not need to repeat this here. Suggest removing the actor examples.

r-c-correa replied:
Of course, I'll make the adjustment 😄

msfussell commented on daprdocs/content/en/reference/components-reference/supported-state-stores/setup-cockroachdb.md:
```diff
@@ -0,0 +1,74 @@
+
+## Create CockroachDB
+
+{{< tabs "Self-Hosted" >}}
```
Can you also provide an example of how to deploy to Kubernetes? Also any other hosting environments if these are easy to reference into the CockroachDB docs (e.g. Deploying to production VM environment)

r-c-correa replied:
Of course, I'll make the adjustment 😃

duglin committed:
Point to the nested tracing doc (#2233)

Just a minor help for readers... take them directly to the doc in question
rather than to the TOC, from which they then need to cick to get to where
they want to be.

Signed-off-by: Doug Davis <dug@microsoft.com>

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

wimvdc committed:
Updated state store docs with DynamoDB TTL (#2203)

* Updated state store docs with DynamoDB TTL

Signed-off-by: Wim <wim.vdc@hotmail.com>

* Making some consistency updates

* Added more realistic example

Signed-off-by: Wim <wim.vdc@hotmail.com>

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Co-authored-by: greenie-msft <56556602+greenie-msft@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

duglin committed:
Fix alignment in kubectl output (#2231)

Sorry, my AR got the better of me :-)

Signed-off-by: Doug Davis <dug@microsoft.com>

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

duglin committed:
Remove sentence about a pict since pict isn't there (#2242)

See #2229 for more info

Fixes #2229

Signed-off-by: Doug Davis <dug@microsoft.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

duglin committed:
Remove sentence about a pict since pict isn't there (#2242)

See #2229 for more info

Fixes #2229

Signed-off-by: Doug Davis <dug@microsoft.com>

dmitsh committed:
how to set content type in state store (#2225)

Signed-off-by: Dmitry Shmulevich <dmitsh@users.noreply.github.com>

Co-authored-by: Dmitry Shmulevich <dmitsh@users.noreply.github.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

yaron2 committed:
update s3 binding (#2246)

Signed-off-by: yaron2 <schneider.yaron@live.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

chiru1221 committed:
Fix grcp metadata in js client (#2243)

Signed-off-by: chiru1221 <takuro1221.it@gmail.com>

chiru1221 committed:
Fix grcp metadata in js client (#2243)

Signed-off-by: chiru1221 <takuro1221.it@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

nwiizo committed:
Update gRPC-integration.md (#2253)

Here, it doesn't work github.com/dapr/go-sdk/dapr/proto/common/v1
It works here github.com/dapr/dapr/pkg/proto/common/v1

nwiizo committed:
Update gRPC-integration.md (#2253)

Here, it doesn't work github.com/dapr/go-sdk/dapr/proto/common/v1
It works here github.com/dapr/dapr/pkg/proto/common/v1

Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

si618 committed:
Fix typo and case inconsistency (#2257)

Signed-off-by: Simon McKenna <sshnug.si+github@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

si618 committed:
Fix typo and case inconsistency (#2257)

Signed-off-by: Simon McKenna <sshnug.si+github@gmail.com>

Commit:
Add clarification on authenticating to Azure (#2241)

* Add clarification on authenticating to Azure

Add a comment pointing further down for more information.

Signed-off-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

* Update to non-depracated fields

Signed-off-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

* Remove extra value

Signed-off-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

Co-authored-by: Mark Fussell <markfussell@gmail.com>

Commit:
Add clarification on authenticating to Azure (#2241)

* Add clarification on authenticating to Azure

Add a comment pointing further down for more information.

Signed-off-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

* Update to non-depracated fields

Signed-off-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

* Remove extra value

Signed-off-by: Aaron Crawfis <Aaron.Crawfis@microsoft.com>

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

msfussell commented:
@r-c-correa PTAL at my suggested changes.

CrazyHZM committed:
Documentation on ACL support for wildcards (#2249)

* acl support wildcard

Signed-off-by: crazyhzm <crazyhzm@apache.org>

* change description

Signed-off-by: crazyhzm <crazyhzm@apache.org>
Signed-off-by: Ricardo Corrêa <r.c.correa@outlook.com>

r-c-correa committed:
Merge branch 'dapr:v1.6' into state_cockroachdb

r-c-correa committed:
Merge branch 'v1.7' into state_cockroachdb

r-c-correa committed:
remove unnecessary actor description and added Kubernetes tab setup for CockroachDB

r-c-correa reviewed (commented):
I made changes in another commit. 😢

msfussell commented:
@r-c-correa - You have pulled in several other changes to the PR. Can you close this PR and open a new ones (or changes this one) to have just the DB Index file and the CockroachDB file. I am unclear on your changes here otherwise.

r-c-correa commented:
I create a new [Pull Request](https://github.com/dapr/docs/pull/2276) @msfussell 😄
